url56b1dc76d3_wufunxiang_Libs_tgz-pJ8-org_msgpack_util_TemplatePrecompilerJ8/src/org/msgpack/template/builder/beans/StandardBeanInfo.java

```diff
@@ -0,0 +1,1505 @@
+         * explicitBeanInfo object:
+         * 1) BeanDescriptor
+         * 2) DefaultEventIndex
+         * 3) DefaultPropertyIndex
+         * These aspects of the beanClass cannot be derived through introspection of the class.
+         * If they are not provided by the explicitBeanInfo, then they must be left null in the 
+         * returned BeanInfo, otherwise they will be copied from the explicitBeanInfo 
+         --------------------------------------------------------------------------------------*/
+        if (explicitBeanInfo != null) {
+            this.explicitBeanInfo = explicitBeanInfo;
+            events = explicitBeanInfo.getEventSetDescriptors();
+            methods = explicitBeanInfo.getMethodDescriptors();
+            properties = explicitBeanInfo.getPropertyDescriptors();
+            defaultEventIndex = explicitBeanInfo.getDefaultEventIndex();
+            if (defaultEventIndex < 0 || defaultEventIndex >= events.length) {
+                defaultEventIndex = -1;
+            }
+            defaultPropertyIndex = explicitBeanInfo.getDefaultPropertyIndex();
+            if (defaultPropertyIndex < 0
+                    || defaultPropertyIndex >= properties.length) {
+                defaultPropertyIndex = -1;
+            }
+            additionalBeanInfo = explicitBeanInfo.getAdditionalBeanInfo();
+
+            if (events != null)
+                explicitEvents = true;
+            if (methods != null)
+                explicitMethods = true;
+            if (properties != null)
+                explicitProperties = true;
+        }
+
+        if (methods == null) {
+            methods = introspectMethods();
+        }
+
+        if (properties == null) {
+            properties = introspectProperties(stopClass);
+        }
+
+        if (events == null) {
+            events = introspectEvents();
+        }
+    }
+
+    @Override
+    public BeanInfo[] getAdditionalBeanInfo() {
+        return null;
+    }
+
+    @Override
+    public EventSetDescriptor[] getEventSetDescriptors() {
+        return events;
+    }
+
+    @Override
+    public MethodDescriptor[] getMethodDescriptors() {
+        return methods;
+    }
+
+    @Override
+    public PropertyDescriptor[] getPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public BeanDescriptor getBeanDescriptor() {
+        if (beanDescriptor == null) {
+            if (explicitBeanInfo != null) {
+                beanDescriptor = explicitBeanInfo.getBeanDescriptor();
+            }
+            if (beanDescriptor == null) {
+                beanDescriptor = new BeanDescriptor(beanClass);
+            }
+        }
+        return beanDescriptor;
+    }
+
+    @Override
+    public int getDefaultEventIndex() {
+        return this.defaultEventIndex;
+    }
+
+    @Override
+    public int getDefaultPropertyIndex() {
+        return this.defaultPropertyIndex;
+    }
+
+    void mergeBeanInfo(BeanInfo beanInfo, boolean force)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 197-208, lines 210-219, lines 221-233 into separate methods.